src/hmem.c

```diff
@@ -41,16 +41,24 @@
 struct ofi_hmem_ops {
 	int (*copy_to_hmem)(void *dest, const void *src, size_t size);
 	int (*copy_from_hmem)(void *dest, const void *src, size_t size);
+
+	/* Return FI_SUCCESS if the HMEM iface owns the buffer. Return
+	 * -FI_ENOSYS if the HMEM iface does not own the buffer. Everything else
```

Comment:
ENOSYS means that the function isn't implemented.  ENODATA is a better option, though this might be okay returning a bool.  If there's an error, we should log it via FI_WARN.  In that case, we can default back to a system buffer, which is most likely.

Reply:
Changing to bool.